### PR TITLE
Tinylog

### LINKED ISSUE #5475: ERROR StatusLogger Log4j2 could not find a logging implementation.
JabRef 5.0.0-dev--2019-10-18----bc26c3ac5
Windows 7 6.1 amd64 
Java 12.0.2

- [+] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. When opening jabref a console starts with this error message
     ERROR StatusLogger Log4j2 could not find a logging implementation. Please add log4j-core to   the classpath. Using SimpleLogger to log to the console...
2. Then Jabref opens still. I do not know what the effect of this error will be. 
3. ...

<details>
  <summary>Log File</summary>
  
  ```
  Paste an excerpt of your log file here
  ```
</details>

### LINKED ISSUE #8007: Replace log4j2 with tinylog2 + slf4j bridge
We always get this Logger error messages on startup and we use a snapshot version of log4j2 which will probably take a long time until it's stable.

We already use slf4j as a Logger interface, so switching to another underlying logger should not be that complicated. Most stuff should be compatible.
We currently use slf4j-2.0.0. This requires at least [tinylog 2.4](https://github.com/tinylog-org/tinylog/pull/199/commits)

https://tinylog.org/v2/download/#slf4j 

## What needs to be changed?

1. Replace log4j2 with tinylog and the bindings.
2. Investigate if we still need our custom Logging Plugins or if they need to be converted  https://tinylog.org/v2/extending/ 
3. Remove log4j stuff also from jpackage/jlink 
4. Test if it works on all platforms

### PR DESCRIPTION
**Pull request by koppor**
Fixes #8007
Fixes #5475 

Debug log level is only possible with java property `-Dtinylog.level=debug `
Tiny log freezes the config after loading

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Does not yet work https://github.com/tinylog-org/tinylog/issues/223

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We trim white space and non-ASCII characters from DOI [#8127](https://github.com/JabRef/jabref/issues/8127)
 - The duplicate checker now inspects other fields in case no difference in the required and optional fields are found.
 - We reworked the library properties dialog and integrated the `Library > Preamble`, `Library > Citation key pattern` and `Library > String constants dialogs` [#8264](https://github.com/JabRef/jabref/pulls/8264)
+- We improved the startup time from JabRef by switching from the logging library `log4j2` to `tinylog` [#8007](https://github.com/JabRef/jabref/issues/8007)
```
Comment (tobiasdiez) [inline CHANGELOG.md:56]:
```suggestion
- We improved the startup time of JabRef by switching from the logging library `log4j2` to `tinylog` [#8007](https://github.com/JabRef/jabref/issues/8007)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -80,13 +81,12 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where XMP and embedded BibTeX export would not work [#8278](https://github.com/JabRef/jabref/issues/8278)
 - We fixed an issue where the XMP and embedded BibTeX import of a file containing multiple schemas failed [#8278](https://github.com/JabRef/jabref/issues/8278)
 - We fixed an issue where pdf-paths and the pdf-indexer could get out of sync [#8182](https://github.com/JabRef/jabref/issues/8182)
-
+- We fixed an issue where Status-Logger error messages appeared on startup of JabRef [#5475](https://github.com/JabRef/jabref/issues/5475)
```
Comment (tobiasdiez) [inline CHANGELOG.md:84]:
```suggestion
- We fixed an issue where Status-Logger error messages appeared during the startup of JabRef [#5475](https://github.com/JabRef/jabref/issues/5475)
```

File: src/main/java/module-info.java [hunk 3]
```diff
@@ -101,4 +99,11 @@
     requires com.fasterxml.jackson.dataformat.yaml;
     requires com.fasterxml.jackson.datatype.jsr310;
     requires net.harawata.appdirs;
+    requires applicationinsights.logging.log4j2;
```
Comment (tobiasdiez) [inline src/main/java/module-info.java:102]: move these things up to the "Logging" section

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
Nice!

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
LGTM

**Pull request merged**